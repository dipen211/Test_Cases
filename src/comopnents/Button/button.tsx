import react, { useState } from 'react';

export const Button = () => {
    const [show, setShow] = useState(true);
    const [count, setCount] = useState(0);
    const [isChecked, setIsChecked] = useState(false);
    const increment = () => {
        setCount(count => count + 1);
    }
    const toggleShow = () => {
        setShow(!show)
    }
    const onChange = () => {
        setIsChecked(!isChecked);
    }
    return (
        <div>
            <button className="buttonText" onClick={toggleShow}>Toggle</button>
            {show && <div>Text</div>}
            <button className="buttonAdd" onClick={increment}>Add</button>
            <h2>{count}</h2>
            <input
                type="checkbox"
                checked={isChecked}
                onChange={onChange}
            />
            {isChecked ? "On" : "Off"}
        </div>
    )
}