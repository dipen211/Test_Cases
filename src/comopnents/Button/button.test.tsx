import react from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button } from './button';
import { mountToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
    let wrapper:any;
    beforeEach(() => {
        wrapper = shallow(<Button />);
    });
    it('should match the snapshot', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    });
    it('should show text', () => {
        const text = wrapper.find('div div');
        expect(text.text()).toBe('Text');
    });
    it('should hide text when button is clicked', () => {
        const button = wrapper.find('button.buttonText');
        button.simulate('click');
        const text = wrapper.find('div div');
        expect(text.length).toBe(0);
    });
    it('should increment number when button is clicked', () => {
        const total = wrapper.find('div h2').text();
        expect(total).toBe("0");
        const button = wrapper.find('button.buttonAdd');
        button.simulate('click');
        const total1 = wrapper.find('div h2').text();
        expect(total1).toBe("1");
    });
    it('should checked when checkbox is clicked', () => {
        expect(wrapper.text()).toEqual('ToggleTextAdd0Off');
        wrapper.find('input').simulate('change');
        expect(wrapper.text()).toEqual('ToggleTextAdd0On');
    });
})