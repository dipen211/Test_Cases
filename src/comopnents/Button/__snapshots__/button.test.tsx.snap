// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<div>
  <button
    className="buttonText"
    onClick={[Function]}
  >
    Toggle
  </button>
  <div>
    Text
  </div>
  <button
    className="buttonAdd"
    onClick={[Function]}
  >
    Add
  </button>
  <h2>
    0
  </h2>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</div>
`;
